### DISCUSSION TITLE
[Application Development] How can I submit a withdrawal proof myself

### DISCUSSION THREAD

**Discussion by threetoeddinosaur** [ethereum-optimism/developers#692] [General]
### Did you check the documentation?

- [X] Yes
- [ ] No

### Documentation Feedback (Optional)

Unclear on where I can find the required details to make the call proveWithdrawalTransaction on OptimismPortal contract. Also unsure if L2OutputOracle is still a valid Contract

### What type of issue are you experiencing?

- [ ] Error
- [ ] Feature Request
- [X] General Question

### Issue Description

I am following the withdrawal flow as given per the documentation but instead of using the sdk for proof submittion, I am submitting the proof myself by making call to this function 
```
function proveWithdrawalTransaction(
        Types.WithdrawalTransaction memory _tx,
        uint256 _disputeGameIndex,
        Types.OutputRootProof calldata _outputRootProof,
        bytes[] calldata _withdrawalProof
    )
```
of OptimismPortal2 Contract.

I am struggling with and how to achieve the values uint256 _disputeGameIndex,Types.OutputRootProof calldata _outputRootProof and bytes[] calldata _withdrawalProof 

### Steps to Reproduce

I called the sendMessage function on L2CrossDomainMessenger which internally calls the initializeWithdrawal function of L2ToL1MessageParser contract 

### Are you using a specific library or SDK?

- [ ] OP-SDK
- [ ] Viem
- [ ] Other
- [ ] Not Sure

### Are you using the latest version of the library/SDK?

- [ ] Yes
- [ ] No

### What type of support do you need?

- [ ] Technical support for issue resolution
- [ ] Feature implementation guidance
- [X] General advice or consultation
- [X] Documentation or learning resources

**Comment by Incede** [comment]
One way to get those params would be to use viem op-stack library. I suspect the following code block in [this](https://gist.github.com/Incede/1ff1358ba93fe8ee53ff5bb08fa21864) script would get you the needed param(s).
```
const proveArgs = await publicClientL2.buildProveWithdrawal({
  output,
  withdrawal,
})
```

**Reply by threetoeddinosaur** [reply]
@Incede Hey thanks, so I am keen on not using any libraries, what is the workaround to do it without any sdks?

**Reply by threetoeddinosaur** [reply]
Also viem op-stack is using L2OutputOracle contract but on testnet it is not in use. So I am confused.

**Comment by joohhnnn** [comment]
Hi there, you can read the source code of VIEM or SDK then build you own script to make the withdraw, there are some clues may give you a help

# L2 to L1 Withdrawal Process

## Contract Methods

### 1. L2 Contract: L2_TO_L1_MESSAGE_PASSER
```solidity
initiateWithdrawal(
    address _target,    // receiving address
    uint256 _gasLimit,  // gas limit
    bytes _data        // additional data
) payable              // withdrawal amount passed via value
```

### 2. L1 Contract: PORTAL
```solidity
proveWithdrawalTransaction(
    WithdrawalTx _tx,          // withdrawal transaction info
    uint256 _l2OutputIndex,    // L2 output index
    OutputRootProof _proof,    // output root proof
    bytes[] _withdrawalProof   // withdrawal proof
)

finalizeWithdrawalTransaction(
    WithdrawalTx _tx           // withdrawal transaction info
)
```

## Process Flow

### 1. Initiate (on L2)
- User calls L2_TO_L1_MESSAGE_PASSER.initiateWithdrawal()
- Locks funds on L2 and emits a MessagePassed event

### 2. Prove (on L1)
- After L2 block is finalized on L1
- User proves the withdrawal by submitting proof to PORTAL contract
- Starts challenge period

### 3. Finalize (on L1)
- After challenge period ends
- User calls PORTAL to finalize and receive funds

These native contract calls complete the withdrawal process from L2 to L1.

**Reply by Padraic-O-Mhuiris** [reply]
To clarify, is there not an additional call to `resolveClaim()` and `resolve()` in the disputeGame? I have a test deployment which use the PermissionedGame as default but it appears by our experiments that this is also required?

**Reply by joohhnnn** [reply]
If you've enabled Fault Proofs, it means you've also activated the op-challenger component. The op-challenger will handle tasks like `resolve()` or `resolveClaim()`operations on your behalf.
